In [3]:
import pandas as pd
import numpy as np
import csv

#### Incidents

In [4]:
df = pd.read_csv('Incident.csv', error_bad_lines=False)
del df['ABCD Misc']

b'Skipping line 18644: expected 135 fields, saw 138\nSkipping line 18980: expected 135 fields, saw 138\nSkipping line 18981: expected 135 fields, saw 138\nSkipping line 18982: expected 135 fields, saw 138\nSkipping line 19020: expected 135 fields, saw 136\nSkipping line 20034: expected 135 fields, saw 137\nSkipping line 20387: expected 135 fields, saw 136\n'
b'Skipping line 23288: expected 135 fields, saw 146\n'
b'Skipping line 25529: expected 135 fields, saw 139\nSkipping line 25534: expected 135 fields, saw 140\n'
b'Skipping line 29461: expected 135 fields, saw 136\n'
b'Skipping line 35414: expected 135 fields, saw 136\nSkipping line 35451: expected 135 fields, saw 137\nSkipping line 35494: expected 135 fields, saw 136\nSkipping line 36184: expected 135 fields, saw 137\n'
b'Skipping line 47145: expected 135 fields, saw 138\n'
b'Skipping line 51522: expected 135 fields, saw 137\nSkipping line 51764: expected 135 fields, saw 136\nSkipping line 52583: expected 135 fields, saw 136\n'
b'S

In [5]:
df.shape

(139859, 134)

#### Resource Table

In [6]:
df2 = pd.read_csv('Incident_ResourceSummary.csv', error_bad_lines=False)

In [7]:
df2.shape

(39837, 16)

So that means 150 total.

In [12]:
for col in df2.columns:
    print(col)

Created On Date Time
Irwin ID
Incident Resource Summary ID
Modified On Date Time
OBJECTID
Resource Agency
ResourceCategory
ResourceKind
ResourcePersonnelQuantity
ResourceProvider
ResourceQuantity
ResourceQuantityCurrentAsOf
ResourceStatus
ResourceType
Created By System
Modified By Sytem


#### John Carr's Incidents

In [8]:
df3 = pd.read_csv('Incident_0.csv', error_bad_lines=False)

//anaconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3058: DtypeWarning: Columns (0,1,4,20,22,23,24,25,26,28,29,31,33,34,37,43,45,47,48,49,50,53,58,59,61,62,63,64,65,66,68,70,71,72,85,86,89,92,95,96,97,98,99,100,102,105,106,107,108,109,110,111,112,113,114,115,116,117,118,120,121,126,127,128,129,130,131,133,134,135,136,137,143,151,153,157,159,160,167,168,169,170,173,181,185,187,188,189,190,191,198) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [9]:
df3.shape

(130801, 201)

In [10]:
df3.describe()

Calculated Acres  Created On Date Time  Critical Resource Needs  \
count       1524.000000          1.308010e+05                      0.0   
mean        7982.104526          1.494881e+12                      NaN   
std        28117.718398          5.081160e+10                      NaN   
min            0.003400          1.400650e+12                      NaN   
25%           72.774525          1.454440e+12                      NaN   
50%          562.435800          1.500070e+12                      NaN   
75%         4019.698900          1.534460e+12                      NaN   
max       496219.602900          1.571780e+12                      NaN   

         Daily Acres  Discovery Acres  Estimated Cost To Date    Fatalities  \
count   90743.000000     87559.000000            1.001200e+04  10259.000000   
mean      533.590953        25.476500            1.220494e+06      0.003412   
std      8078.943998      1089.125036            7.794344e+06      0.081944   
min         0.000000         0.000000            0.000000e+00      0.000000   
25%         0.100000         0.100000            1.000000e+03      0.000000   
50%         0.250000         0.100000            1.000000e+04      0.000000   
75%         5.000000         0.500000            1.795778e+05      0.000000   
max    779292.000000    250000.000000            2.625000e+08      4.000000   

         Final Acres  Final Fire Report Approved By  \
count    7374.000000                            0.0   
mean      394.090291                            NaN   
std      5339.945024                            NaN   
min         0.000000                            NaN   
25%         0.100000                            NaN   
50%         1.000000                            NaN   
75%         8.000000                            NaN   
max    318156.000000                            NaN   

       Final Fire Report Approved By Title  ...  Unnamed: 184  Unnamed: 186  \
count                                  0.0  ...           1.0           0.0   
mean                                   NaN  ...          28.0           NaN   
std                                    NaN  ...           NaN           NaN   
min                                    NaN  ...          28.0           NaN   
25%                                    NaN  ...          28.0           NaN   
50%                                    NaN  ...          28.0           NaN   
75%                                    NaN  ...          28.0           NaN   
max                                    NaN  ...          28.0           NaN   

       Unnamed: 192  Unnamed: 193  Unnamed: 194  Unnamed: 195  Unnamed: 196  \
count           0.0           0.0           0.0           0.0           0.0   
mean            NaN           NaN           NaN           NaN           NaN   
std             NaN           NaN           NaN           NaN           NaN   
min             NaN           NaN           NaN           NaN           NaN   
25%             NaN           NaN           NaN           NaN           NaN   
50%             NaN           NaN           NaN           NaN           NaN   
75%             NaN           NaN           NaN           NaN           NaN   
max             NaN           NaN           NaN           NaN           NaN   

       Unnamed: 197  Unnamed: 199  Unnamed: 200  
count           0.0       1.00000       1.00000  
mean            NaN     -84.29389      37.04889  
std             NaN           NaN           NaN  
min             NaN     -84.29389      37.04889  
25%             NaN     -84.29389      37.04889  
50%             NaN     -84.29389      37.04889  
75%             NaN     -84.29389      37.04889  
max             NaN     -84.29389      37.04889  

[8 rows x 74 columns]

In [11]:
for col in df3.columns:
    print(col)

ABCD Misc
Additional Fuel Model
ADS Permission State
Calculated Acres
Conflict Parent Irwin ID
Containment Date Time
Control Date Time
Created On Date Time
Critical Resource Needs
Daily Acres
Discovery Acres
Dispatch Center ID
Estimated Containment Date
Estimated Cost To Date
Fatalities
Final Acres
Final Fire Report Approved By
Final Fire Report Approved By Title
Final Fire Report Approved By Unit
Final Fire Report Approved Date
Final Fire Report Narrative
Final Strategy Attained Date Time
Fire Behavior Description
Fire Behavior General
Fire Behavior General 1
Fire Behavior General 2
Fire Behavior General 3
Fire Cause
Fire Cause General
Fire Cause Specific
Fire Code
Fire Department ID
Fire Discovery Date Time
Fire Growth Cessation Date Time
Fire Ignition Date Time
Fire Mgmt Complexity
Fire Out Date Time
Fire Strategy Confine Percent
Fire Strategy Full Supp Percent
Fire Strategy Monitor Percent
Fire Strategy Point Zone Percent
Fiscally Responsible Unit
FS Job Code
FS Override Code
Geogr

We have the same number of usable features that John Carr did. In general, he never really ever had 201 features. 

#### Determine number of Naan values

In [28]:
df.isnull().sum()

Additional Fuel Model              138769
ADS Permission State                    0
Calculated Acres                   138319
Conflict Parent Irwin ID           139847
Containment Date Time               55192
                                    ...  
Strategic Decision Publish Date    139389
WFDSS Strategy Slider Value        139816
GlobalID                                0
x                                       0
y                                       0
Length: 134, dtype: int64

In [32]:
df.isnull().sum()/len(df)*100

Additional Fuel Model              99.220644
ADS Permission State                0.000000
Calculated Acres                   98.898891
Conflict Parent Irwin ID           99.991420
Containment Date Time              39.462602
                                     ...    
Strategic Decision Publish Date    99.663947
WFDSS Strategy Slider Value        99.969255
GlobalID                            0.000000
x                                   0.000000
y                                   0.000000
Length: 134, dtype: float64

In [44]:
Fireinfo_clean = df.dropna(axis = 1, thresh = len(df)*.15)

In [45]:
Fireinfo_clean.head()

ADS Permission State  Containment Date Time      Control Date Time  \
0              DEFAULT  6/13/2014 12:17:00 AM  6/13/2014 12:17:00 AM   
1              DEFAULT  6/15/2014 10:00:00 AM  6/15/2014 11:00:00 AM   
2              DEFAULT   6/13/2014 4:00:00 PM  6/14/2014 12:00:00 AM   
3              DEFAULT  6/14/2014 12:30:00 AM  6/14/2014 12:30:00 AM   
4              DEFAULT   6/13/2014 4:00:00 AM   6/13/2014 3:00:00 AM   

   Created On Date Time  Daily Acres  Discovery Acres Dispatch Center ID  \
0         1402618753000          NaN              NaN                NaN   
1         1402825466000          0.2              NaN                NaN   
2         1402619601000       1200.0              NaN                NaN   
3         1402848698000          0.1              NaN                NaN   
4         1402630436000          0.2              NaN                NaN   

  Fire Cause Fire Code Fire Discovery Date Time  ... Unique Fire Identifier  \
0        NaN      H5W3     6/12/2014 9:27:00 PM  ...      2014-UTVLD-000088   
1      Human      H52L     6/15/2014 8:00:00 AM  ...      2014-CASRF-000784   
2      Human      H5W7    6/12/2014 10:18:00 PM  ...      2014-ID2CX-000222   
3      Human       NaN    6/13/2014 11:05:00 PM  ...      2014-UTSES-004078   
4        NaN      H5XD     6/13/2014 2:33:00 AM  ...      2014-AZSCA-000088   

  WFDSS Decision Status Created By System Modified By System  \
0                   NaN          firecode              wfdss   
1                   NaN          firecode              wfdss   
2                   NaN          firecode              wfdss   
3                   NaN             wfdss              wfdss   
4                   NaN          firecode              wfdss   

              Created On           Modified On Is Dispatch Complete  \
0  6/13/2014 12:19:13 AM   7/8/2014 9:07:19 PM                  NaN   
1   6/15/2014 9:44:26 AM   7/8/2014 7:06:35 PM                  NaN   
2  6/13/2014 12:33:21 AM   8/9/2014 9:19:27 PM                  NaN   
3   6/15/2014 4:11:38 PM  6/15/2014 4:43:28 PM                  NaN   
4   6/13/2014 3:33:56 AM   7/9/2014 7:02:27 PM                  NaN   

                               GlobalID        x        y  
0  a5d6955a-e92e-48ef-94e7-e40ce22721c9 -109.456  40.5942  
1  a09e8248-422a-4014-b295-a8f301f96fc5 -123.531  41.3083  
2  71ee890a-71e9-4b20-8f91-f5a344af3ff7 -116.512  43.2944  
3  988adcc7-d82b-44cf-bfa7-36c788229245 -110.791  39.5864  
4  f171f030-3dab-47a6-93ef-2682a21e52b4 -110.454  33.3819  

[5 rows x 54 columns]

In [46]:
Fireinfo_clean.shape

(139859, 54)

This means that there are at most 15% of null values for each column.

In [47]:
for col in df.columns:
    print(col)

Additional Fuel Model
ADS Permission State
Calculated Acres
Conflict Parent Irwin ID
Containment Date Time
Control Date Time
Created On Date Time
Critical Resource Needs
Daily Acres
Discovery Acres
Dispatch Center ID
Estimated Containment Date
Estimated Cost To Date
Fatalities
Final Acres
Final Fire Report Approved By
Final Fire Report Approved By Title
Final Fire Report Approved By Unit
Final Fire Report Approved Date
Final Fire Report Narrative
Final Strategy Attained Date Time
Fire Behavior Description
Fire Behavior General
Fire Behavior General 1
Fire Behavior General 2
Fire Behavior General 3
Fire Cause
Fire Cause General
Fire Cause Specific
Fire Code
Fire Department ID
Fire Discovery Date Time
Fire Growth Cessation Date Time
Fire Ignition Date Time
Fire Mgmt Complexity
Fire Out Date Time
Fire Strategy Confine Percent
Fire Strategy Full Supp Percent
Fire Strategy Monitor Percent
Fire Strategy Point Zone Percent
Fiscally Responsible Unit
FS Job Code
FS Override Code
Geographic Area

columns = ['ADS Permission State', 'Calculated Acres', 'Daily Acres','Discovery Acres', 'Estimated Cost To Date', 'Fatalities','Final Acres', 'Fire Cause', 
 'Fire Cause General','Fire Cause Specific', 'Fire Code', 'Fire Discovery Date Time', 'Fire Out Date Time', 'Geographic Area Coordination Center Code', 
 'Has Fatalities', 'Has Injuries', 'Incident Type Category', 'Incident Type Kind', 'Injuries', 'Irwin ID', 'Is Fire Cause Investigated', 
 'Is Fire Code Requested', 'Is FS Assisted', 'Is Initial Fire Strategy Met', 'Is Quarantined', 'Is Reimbursable',
 'Is Trespass', 'POO Landowner Category', 'POO Protecting Unit', 'POO State', 
 'Residences Destroyed', 'Residences Threatened', 'Estimated Final Cost', 'x', 'y']

In [48]:
df.nunique()

Additional Fuel Model                  13
ADS Permission State                    3
Calculated Acres                     1534
Conflict Parent Irwin ID               12
Containment Date Time               71633
                                    ...  
Strategic Decision Publish Date       470
WFDSS Strategy Slider Value            27
GlobalID                           139859
x                                  112489
y                                  124169
Length: 134, dtype: int64

In [49]:
# Drop records that don't have lat/long and an ID number
df.dropna(subset=['x', 'y', 'Irwin ID'], inplace=True)